File: UPG1/set10/fact.cpp
#include <iostream>

using std::cin;
using std::cout;
using std::endl;

void solve(long long const n, long long const m)
{
	long long p = 1;
	for (long long i = 2; i <= n and p != 0 and m != 0; ++i)
	{
		p = (p * i) % m;
	}
	if (p <= 0) cout << m << " divides " << n << "!\n";
	else cout << m << " does not divide " << n << "!\n";
}

int main()
{
	long long n,m;
	while (cin >> n >> m)
	{
		solve(n,m);
	}
	cout << endl;
}

File: UPG1/set10/fact.py
def g(n):
    l=[]
    for i in range(1,n+1): 
        if n%i<1:l+=[i]
    return l
*f,= map(str.split,open(0).read().splitlines())
s=" divide"
for n,m in f:
    n=int(n);m=int(m)
    if set(g(n)).issubset(set(g(m))):print(f"{m}{s}s {n}!")
    else:print(f"{m} does not{s} {n}!")
